pkg/unifiedpush/client.go

```diff
@@ -124,6 +195,53 @@ func (c UnifiedpushClient) CreateAndroidVariant(av *pushv1alpha1.AndroidVariant)
 	return createdVariant.VariantId, nil
 }
 
+// DeleteAndroidVariant deletes an Android variant in UPS
+func (c UnifiedpushClient) DeleteAndroidVariant(v *pushv1alpha1.AndroidVariant) error {
+	if v.Status.VariantId == "" {
+		// We haven't created it yet
+		return nil
```

Comment:
Not sure if this should return an error. I'm not convinced that an error would be useful.